skore/src/skore/cross_validate.py

```diff
@@ -0,0 +1,188 @@
+    from sklearn.base import is_classifier, is_regressor
+
+    if y is None:
+        # FIXME: The task might not be clustering
+        return "clustering"
+
+    if is_classifier(estimator):
+        return "classification"
+
+    if is_regressor(estimator):
+        return "regression"
+
+    type_of_target = sklearn.utils.multiclass.type_of_target(y)
+
+    if type_of_target == "unknown":
+        return "unknown"
+
+    if "continuous" in type_of_target:
+        return "regression"
+
+    return "classification"
+
+
+def _expand_scorers(scorers, scorers_to_add: list[str]):
+    """Expand `scorers` with `scorers_to_add`.
+
+    Parameters
+    ----------
+    scorers : any type that is accepted by scikit-learn's cross_validate
+        The scorer(s) to expand.
+    scorers_to_add : list[str]
+        The scorers to be added
+
+    Returns
+    -------
+    new_scorers : dict[str, str | None]
+        The scorers after adding `scorers_to_add`.
+    added_scorers : list[str]
+        The scorers that were actually added (i.e. the ones that were not already
+        in `scorers`).
+    """
+    added_scorers = []
+
+    if scorers is None:
+        new_scorers = {"score": None}
+        for s in scorers_to_add:
+            new_scorers[s] = s
+            added_scorers.append(s)
+    elif isinstance(scorers, str):
+        new_scorers = {"score": scorers}
+        for s in scorers_to_add:
+            if s == scorers:
+                continue
+            new_scorers[s] = s
+            added_scorers.append(s)
+    elif isinstance(scorers, dict):
+        new_scorers = scorers.copy()
+        for s in scorers_to_add:
+            if s in scorers:
+                continue
+            new_scorers[s] = s
+            added_scorers.append(s)
+    elif isinstance(scorers, list):
+        new_scorers = scorers.copy()
+        for s in scorers_to_add:
+            if s in scorers:
+                continue
+            new_scorers.append(s)
+            added_scorers.append(s)
+    elif isinstance(scorers, tuple):
+        scorers = list(scorers)
+        new_scorers, added_scorers = _expand_scorers(scorers, scorers_to_add)
+
+    return new_scorers, added_scorers
+
+
+def _strip_cv_results_scores(cv_results, added_scorers):
+    """Remove information about `added_scorers` in `cv_results`."""
+    # Takes care both of train and test scores
+    return {
+        k: v
+        for k, v in cv_results.items()
+        if not any(added_scorer in k for added_scorer in added_scorers)
+    }
+
+
+def cross_validate(
+    *args, project: Project | None = None, **kwargs
+) -> CrossValidationItem:
+    """Evaluate estimator by cross-validation and output UI-friendly object.
+
+    This function wraps scikit-learn's [cross_validate](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_validate.html#sklearn.model_selection.cross_validate)
+    function, to provide more context
+    and facilitate the analysis.
```

Comment:
Maybe keep it on one line?

Reply:
That would lead to ruff complaining that the line is too long. If this helps, in markdown new lines within paragraphs are usually discarded (although not in github-flavoured markdown apparently).